pydicom/filewriter.py

```diff
@@ -579,7 +590,7 @@ def write_file_meta_info(fp, file_meta, enforce_standard=True):
         If any non-Group 2 Elements are present in `file_meta`.
     """
     # Check that no non-Group 2 Elements are present
-    for elem in file_meta:
+    for elem in file_meta.raw():
```

Comment:
Needs test coverage

Reply:
Actually this is covered by the tests that read a dataset and write it afterwards, though not explicitely regarding the implementation.

Reply:
I'd be happier if there were explicit tests for the changes since the tests usually aren't written with them in mind.